torch/fx/passes/split_module.py

```diff
@@ -289,6 +386,25 @@ def instantiate_node_partition_mapping(node):
             new_node.meta = node.meta.copy()
             partition.environment[node] = new_node
 
+    # Exit epilogue
+    for regions_mapping in [autocast_regions]:
+        # we need to unwind in reverse order to setting up
+        for node in reversed(regions_mapping):
+            regions = regions_mapping[node]
+            assert len(regions) > 0
+            for r in regions[:-1]:
+                partition = partitions[str(r)]
+                exit_node = autocast_exits[node]
+                assert exit_node is not None, "Missing exit node"
+                new_node = partition.graph.create_node(
```

Comment:
In particular @wconstab if we try to record cross partition dependencies on the old graph rather than new graph, this replacement of the exit node's argument to the subgraph's copy of the enter is not performed.